Release workflow

frankinspace opened the pull request:
I was able to test everything up to publishing to npm. It is trying to do automatic changelog tracking as well.

There are two steps involved:
prepare release
finalize release

You run prepare first and then finalize when ready. This is the best documentation I've found https://jupyter-releaser.readthedocs.io/en/latest/how_to_guides/convert_repo_from_repo.html#initial-release-workflow

Also, this blog post: https://blog.jupyter.org/automate-your-releases-with-the-jupyter-releaser-701e7b9841e6

But it is still a bit lacking. Will need to setup the trusted publisher once merged

Commit:
Start build pipeline and bump jupyterlab to 4.2.5 and react to 18.2

Commit:
only need to test ubuntu

Commit:
accidentally removed lock file

Commit:
bump jupyter version in build container

Commit:
trying build

Commit:
disable cicd app in fork

Commit:
disable lint in fork

Commit:
install before test

Commit:
use working build

Commit:
run test

Commit:
run isolated test of wheel

Commit:
update playwright tests

Commit:
update playwright tests

Commit:
split up UI tests

Commit:
split up UI tests

Commit:
split up UI tests

Commit:
split up UI tests

Commit:
split up UI tests

Commit:
rename config

Commit:
add other workflows

Commit:
Merge branch 'issues/Community/1050' into develop

Commit:
add changelog

Commit:
fix action verison

Commit:
fix app id

Commit:
try publish

Commit:
try publish

frankinspace committed:
Publish 0.3.7

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.7-py3-none-any.whl: 1e16dc45838abe6deb49b33b5cd2d098e99fecdc906ed1541c661bde455072be

maap_algorithms_jupyter_extension-0.3.7.tar.gz: 632daa8d42229142c4429332ba4c05c8b7a666db4db3d8d7764999f1471cb4cc

maap_algorithms_jupyter_extension-0.3.7.tgz: 2de1ffe303a40ea417d5e2223a7c333c235dd1dd8c3af5a3ba11882e2cc5dbc0

Commit:
try publish

frankinspace committed:
Publish 0.3.8

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.8-py3-none-any.whl: d1e468bf933f250f6526b1bc3f71208d3dbb392ca0e50d935960b56cef5146e2

maap_algorithms_jupyter_extension-0.3.8.tar.gz: 70720ecc0c569830cee6a6f7ae11f3f722ac8c6056695074919fb8136be2d4bf

maap_algorithms_jupyter_extension-0.3.8.tgz: 893e6d009ca3962600e3c95e37ddbe176086b12e68ab3fe3c8d5c106afbf8841

Commit:
try publish

frankinspace committed:
Publish 0.3.9

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.9-py3-none-any.whl: baf0833d3c015747f9c00c52381be7477d026d87b121eb9bfc582911e06ee977

maap_algorithms_jupyter_extension-0.3.9.tar.gz: 5f0f49f938b6a56d16162dd81795ac8f8d7e877f6aa947cea1b1a48f0ce2693a

maap_algorithms_jupyter_extension-0.3.9.tgz: 59cb8e4109d38275ed7dc26531ed35c65f25306cdf8653b469124e72c4039ccc

Commit:
try publish

frankinspace committed:
Publish 0.3.10

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.10-py3-none-any.whl: 04172a95512462897ffa4a0d1a68bd87810f1c5c5ad5ae95954a36b948fe571f

maap_algorithms_jupyter_extension-0.3.10.tar.gz: d0661c4e51534752679a65f3745ebe937318604532ba2ba29c4dbb461fd4923b

maap_algorithms_jupyter_extension-0.3.10.tgz: 67c19778720277e0916ac096e3c188e51b2d0391cd8a7deab936970807dac0fa

frankinspace committed:
Publish 0.3.11

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.11-py3-none-any.whl: 1d084a6c8fdb756f0ab1fcf6677138aeb393361b5f6923db282d0e683a39b81c

maap_algorithms_jupyter_extension-0.3.11.tar.gz: c7ac969491633fb65f64796d1207ad97c5bcbe34b847949cba5f2cfe78386509

maap_algorithms_jupyter_extension-0.3.11.tgz: 8efb65e0bd2b1f2688df876fd225cab1aef156276ce2b4edf6e3830dbc82d969

Commit:
try publish

frankinspace committed:
Publish 0.3.12

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.12-py3-none-any.whl: ca8b5be029a54510a356e6761c087c0beaad3f13d6594036b90cfc4cc8b3ebac

maap_algorithms_jupyter_extension-0.3.12.tar.gz: db72541fb8c72f7a6e09cd23e3e8b3f59867177bc218909edc3e725dcdf5fd0b

maap_algorithms_jupyter_extension-0.3.12.tgz: e7e7629ea054c1e51331345971b0df2c3e92bbdebefea8237c47a06e9d6d421b

frankinspace committed:
Publish 0.3.13

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.13-py3-none-any.whl: 067347c8e2c5b5a94c1f890b689c63627dcad9ea45d4fd8d27d4f0741e3f17f5

maap_algorithms_jupyter_extension-0.3.13.tar.gz: 3e9a53060d18b601d7e25db2bf4d5a3457e640c3f4a78a82684321c7f0b7c265

maap_algorithms_jupyter_extension-0.3.13.tgz: c1bbfe0830f2a9cfb6c774336cb1dc6d188c01afea46c6b3c6442b79001037b6

Commit:
try publish

frankinspace committed:
Publish 0.3.14

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.14-py3-none-any.whl: ab99181d406ab5bfc08369b49db36cbacd6bf5b3462e196db4cfb97330b90692

maap_algorithms_jupyter_extension-0.3.14.tar.gz: cf91c26c1cd43b4dae0f07bc42ed0e7e1f9a547dcc07e9e09f65b5f5873fb4a1

maap_algorithms_jupyter_extension-0.3.14.tgz: 1264aff60b0d0ef9e220b237a614cb25f4d323ea94dd67eef0e7502d2bb386c4

Commit:
try publish

Commit:
try publish

frankinspace commented:
@sujen1412 I'm going to be out for a bit but this is what I have so far

Commit:
merge upstream

Commit:
reset package version

marjo-luc commented on .github/workflows/build.yml:
Does the timeout need to be that long? Wondering if 2-5 min would be sufficient.

frankinspace replied:
not sure what timeout this is referring to

marjo-luc replied:
Line 184:
`timeout-minutes: 15`

marjo-luc commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,188 @@
+      - name: Install dependencies
+        run: python -m pip install -U "jupyterlab>=4.0.0,<5"
+
+#      - name: Lint the extension
```
Is this commented out because the results are scary?

frankinspace replied:
yeah pretty much. It also requires installing a linter in the dependencies so I think would be handled better in it's own update (adding a linter and fixing linter problems)

marjo-luc reviewed (approved):
This looks good to me. I ran through part of this workflow locally to make sure the test workflow ran as expected and the expected artifacts were generated. I'd like to run through a full release within this milestone just to confirm everything looks as expected.

Also, we already discussed making npm publication optional in our previous hackathon since we aren't really publishing these packages to npm, but adding that note here for posterity.

frankinspace committed:
Update publish-release.yml

Commit:
change url back for test

Commit:
add to .gitignore

Commit:
advance version for testing

frankinspace committed:
Publish 0.3.15

SHA256 hashes:

maap_algorithms_jupyter_extension-0.3.15-py3-none-any.whl: 1e285bf91457cadc10b43ccfbd9656247a8450ad2fd57f6bcf8e1c37c07f7433

maap_algorithms_jupyter_extension-0.3.15.tar.gz: d6ae2772f84f410c8fd8528fae95eebdefb74991a99043465ef6c6211128d2e3

maap_algorithms_jupyter_extension-0.3.15.tgz: ae6b4de945f11c6025d4c28a19bb98fe9fd4aa460a286a8cc87e258e289784fc

Commit:
reset for merge

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

# Conflicts:
#	package.json
#	yarn.lock

Commit:
Add Material UI to dependencies because it looks like it is used in src/components/InputRow.tsx and src/components/TableConfigInputs.tsx
